video2dataset/downloader.py

```diff
@@ -9,17 +9,19 @@
 
 def handle_youtube(youtube_url):
     """returns file and destination name from youtube url."""
-    ydl_opts = {}
+    yt_dlp.utils.std_headers['Referer'] = "https://www.youtube.com/"
+    yt_dlp.utils.std_headers['User-Agent'] = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36'
+
+    ydl_opts = {
+        'quiet': True,
+        'format': 'bv*[height<=360][ext=mp4]+ba[ext=m4a]/b[height<=360]'
+    }
+
     ydl = yt_dlp.YoutubeDL(ydl_opts)
     info = ydl.extract_info(youtube_url, download=False)
     formats = info.get("formats", None)
-    f = None
-    for f in formats:
-        if f.get("format_note", None) != QUALITY:
-            continue
-        break
-
-    cv2_vid = f.get("url", None)
+    cv2_vid = [f for f in formats if f['format_note']
```

Comment:
you're still looping over the formats here and not using your "format" specification. What you want is to look through 'requested_formats'

Reply:
Yeah, I fixed this